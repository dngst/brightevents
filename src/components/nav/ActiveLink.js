import { Link, useRoute } from "wouter";

const ActiveLink = (props) => {
  const [isActive] = useRoute(props.href);
  return (
    <Link {...props}>
      <a className={isActive ? "active" : ""} href={isActive}>
        {props.children}
      </a>
    </Link>
  );
};

export default ActiveLink;
