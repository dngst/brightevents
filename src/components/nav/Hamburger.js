import styled from "styled-components";

const Hamburger = styled.div`
  display: none;
  flex-direction: column;
  cursor: pointer;
  span {
    height: 2px;
    width: 25px;
    background: ${(props) => props.theme.base};
    margin-bottom: 4px;
    border-radius: 0;
  }
  @media (max-width: 768px) {
    display: flex;
  }
`;

export default Hamburger;
