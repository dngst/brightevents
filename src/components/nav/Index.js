import styled from "styled-components";

const Index = styled.span`
  padding: 1rem 0;
  cursor: pointer;
  img {
    height: 2rem;
    width: 100%;
  }
`;

export default Index;
