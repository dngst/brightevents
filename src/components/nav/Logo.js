import styled from "styled-components";

const Logo = styled.span`
  padding: 1rem 0;
  cursor: pointer;
  img {
    height: 2rem;
  }
`;

export default Logo;
