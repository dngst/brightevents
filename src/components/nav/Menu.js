import styled from "styled-components";

const Menu = styled.div`
  display: flex;
  justify-content: space-between;
  position: relative;
  a {
    padding: 1rem 2rem;
    cursor: pointer;
    text-align: right;
    text-transform: capitalize;
    color: #000000;
    transition: all 0.3s ease-in;
    font-size: 0.9rem;
    &:hover {
      text-underline-position: under;
      text-decoration: underline;
      text-underline-offset: 0.5em;
      text-decoration-color: ${(props) => props.theme.base};
    }
  }
  @media (max-width: 768px) {
    overflow: hidden;
    flex-direction: column;
    max-height: ${({ isOpen }) => (isOpen ? "300px" : "0")};
    transition: max-height 0.3s ease-in;
    width: 100%;
  }
`;

export default Menu;
