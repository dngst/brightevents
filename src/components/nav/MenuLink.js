import styled from "styled-components";

const MenuLink = styled.a`
  padding: 1rem 2rem;
  cursor: pointer;
  text-align: right;
  text-transform: capitalize;
  color: #000000;
  transition: all 0.3s ease-in;
  font-size: 0.9rem;
  &:hover {
    text-underline-position: under;
    text-decoration: underline;
    text-underline-offset: 0.5em;
    text-decoration-color: #f78f1e;
  }
`;

export default MenuLink;
