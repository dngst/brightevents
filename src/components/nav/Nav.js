import styled from "styled-components";

const Nav = styled.div`
  padding: 0 2rem;
  display: flex;
  justify-content: space-between;
  align-items: center;
  flex-wrap: wrap;
  background: white;
  position: absolute;
  top: 0;
  left: 0;
  right: 0;
  @media (max-width: 768px) {
    border-bottom: 1px solid #f2f2f2;
  }
`;

export default Nav;
