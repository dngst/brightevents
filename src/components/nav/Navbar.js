import React, { useState } from "react";
import Nav from "./Nav";
import Index from "./Index";
import Menu from "./Menu";
import MenuLink from "./MenuLink";
import Hamburger from "./Hamburger";
import Logo from "./logo.png"


const Navbar = () => {
  const [isOpen, setIsOpen] = useState(false);

  return (
    <Nav>
      <Index href="/">
        <img src={Logo} alt="Home" />
      </Index>
      <Hamburger onClick={() => setIsOpen(!isOpen)}>
        <span />
        <span />
        <span />
      </Hamburger>
      <Menu isOpen={isOpen}>
        <MenuLink href="/about">about</MenuLink>
        <MenuLink href="/events">browse events</MenuLink>
        <MenuLink href="/login">login</MenuLink>
      </Menu>
    </Nav>
  );
};

export default Navbar;
