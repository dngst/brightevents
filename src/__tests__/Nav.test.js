import { render } from "@testing-library/react";
import { Nav } from "components/nav/Navbar";

it("works", () => {
  const { container } = render(<Nav />);
  expect(container.firstChild).toMatchSnapshot();
});

it("applies default styles", () => {
  const { container } = render(<Nav />);
  expect(container.firstChild).toHaveStyleRule("position", "absolute");
  expect(container.firstChild).toHaveStyleRule("display", "flex");
  expect(container.firstChild).toHaveStyleRule("justify-content", "space-between");
});
