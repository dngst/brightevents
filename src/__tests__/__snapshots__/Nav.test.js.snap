// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`works 1`] = `
.c0 {
  padding: 0 2rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  position: absolute;
  top: 0;
  left: 0;
  right: 0;
}

.c0 .active {
  text-underline-position: under;
  -webkit-text-decoration: underline;
  text-decoration: underline;
  text-underline-offset: 0.5em;
}

@media (max-width:768px) {
  .c0 {
    border-bottom: 1px solid;
  }
}

<div
  class="c0"
/>
`;
